src/sentry/static/sentry/app/views/eventsV2/sortLink.tsx

```diff
@@ -7,70 +7,70 @@ import {omit} from 'lodash';
 import InlineSvg from 'app/components/inlineSvg';
 import Link from 'app/components/links/link';
 
+import EventView, {Field, Sort, isFieldSortable} from './eventView';
+import {MetaType} from './utils';
+
 type Alignments = 'left' | 'right' | undefined;
 
 type Props = {
-  title: string;
-  sortKey: string;
-  defaultSort: string;
-  location: Location;
   align: Alignments;
+  field: Field;
+  location: Location;
+  eventView: EventView;
+  tableDataMeta: MetaType;
 };
 
 class SortLink extends React.Component<Props> {
   static propTypes = {
     align: PropTypes.string,
-    title: PropTypes.string.isRequired,
-    sortKey: PropTypes.string.isRequired,
-    defaultSort: PropTypes.string.isRequired,
+    field: PropTypes.object.isRequired,
     location: PropTypes.object.isRequired,
+    eventView: PropTypes.object.isRequired,
+    tableDataMeta: PropTypes.object.isRequired,
   };
 
-  getCurrentSort(): string {
-    const {defaultSort, location} = this.props;
-    return typeof location.query.sort === 'string' ? location.query.sort : defaultSort;
-  }
+  isCurrentColumnSorted = (): Sort | undefined => {
+    const {eventView, field, tableDataMeta} = this.props;
 
-  getSort() {
-    const {sortKey} = this.props;
-    const currentSort = this.getCurrentSort();
+    return eventView.isFieldSorted(field, tableDataMeta);
+  };
 
-    // Page is currently unsorted or is ascending
-    if (currentSort === `-${sortKey}`) {
-      return sortKey;
-    }
+  getTarget() {
+    const {location, field, eventView, tableDataMeta} = this.props;
 
-    // Reverse direction
-    return `-${sortKey}`;
-  }
+    const nextEventView = eventView.sortOnField(field, tableDataMeta);
+    const queryStringObject = nextEventView.generateQueryStringObject();
 
-  getTarget() {
-    const {location} = this.props;
     return {
-      pathname: location.pathname,
-      query: {...location.query, sort: this.getSort()},
+      ...location,
+      query: queryStringObject,
     };
   }
 
   renderChevron() {
-    const currentSort = this.getCurrentSort();
-    const {sortKey} = this.props;
-    if (!currentSort || currentSort.indexOf(sortKey) === -1) {
+    const currentSort = this.isCurrentColumnSorted();
+
+    if (!currentSort) {
       return null;
     }
 
-    if (currentSort[0] === '-') {
+    if (currentSort.kind === 'desc') {
       return <InlineSvg src="icon-chevron-down" />;
     }
 
     return <InlineSvg src="icon-chevron-up" />;
   }
 
   render() {
-    const {align, title} = this.props;
+    const {align, field, tableDataMeta} = this.props;
+
+    if (!isFieldSortable(field, tableDataMeta)) {
+      return <span>{field.title}</span>;
```

Comment:
Should non sortable columns also respect alignment?

Reply:
@markstory good catch 👍